Stable Release 1.0 --> Switched to Beta Version 0.2

rayapetkova opened the pull request.

karastoyanov committed:
Backend script for handling the database updated:
 - New category functionality implemented
 - New subcategory functionality implemented

karastoyanov committed:
Basic Main Menu Windows

karastoyanov committed:
Portfolio script update

rayapetkova committed:
Update manage_portfolio

Commented some of the code and made small changes in the queries.

karastoyanov committed:
Main Menu updated

rayapetkova committed:
Implemented the new column

Implemented the new column "unit_of_measure" in the script

karastoyanov committed:
Main Menu rework

karastoyanov committed:
Main Menu Reworked

rayapetkova committed:
Update left_buttons_groupbox

Made a left buttons group box and added a layout in it. Changed the for loop to a while loop... because we pop elements and when there are none left, the while loop will stop.

rayapetkova committed:
Added new group box

New group box, new_layout and two buttons "Favourites" and "Log out"

rayapetkova committed:
Update with the image_url column

Added an image_url column in the table "products" and added an input line saying "Enter an image url: ". Then I added the user input to the query.

karastoyanov commented on db_handle/manage_portfolio.py:
```diff
@@ -0,0 +1,112 @@
+    """Enter the category name"""
+    category_name = input("Enter category name: ")
+
+    # """Calculates the total sub-categories / for new categories is 0 by default"""
```
@rayapetkova We won't use this section with the comments. Could you remove this block of code from the function.

karastoyanov commented on db_handle/manage_portfolio.py:
```diff
@@ -0,0 +1,112 @@
+    """Enter the sub-category name"""
+    subcategory_name = input("Enter the new subcategory name: ")
+
+    # """Calculates the total products inside the sub-category / for new sub-categories is 0 by default"""
```
@rayapetkova We won't use this section with the comments. Could you remove this block of code from the function.

rayapetkova replied:
Should I remove only row 42 or remove all four - 39, 40, 41 and 42?

karastoyanov replied:
Rows from 42 to 44, only the currently commented lines. Same for above function `create_new_category`

karastoyanov reviewed (changes requested):
Check the comments and take actions.

rayapetkova committed:
Removed unnecessary code in manage_portfolio

Removed some commented code

rayapetkova commented:
Last changes: d5526c6
- Removed rows 19-26 and 41-44 inclusive

rayapetkova committed:
Update main_menu

Moved the buttons "Favourites" and "Log out" to the right of the top_buttons_layout.

rayapetkova committed:
Made a few changes in main_menu

Added a basic structure of the creating the categories layout.

karastoyanov committed:
README file draft

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/market-app into development

karastoyanov committed:
Update README.md

karastoyanov committed:
Update README.md

Images added

karastoyanov committed:
Categories Grid Update

karastoyanov committed:
Categories menu update

karastoyanov committed:
Buttons in GridLayout Updated - almost done

karastoyanov committed:
Merge branch 'development' into main_menu

karastoyanov committed:
Delete .obisidan directory

Directory contains only configs, it's not needed in the remote branch.

techfrapi committed:
about window created

techfrapi committed:
change windows size

karastoyanov committed:
Calling buttons fixed

Co-authored-by: rayapetkova rayapetkova12321@gmail.com

karastoyanov committed:
Main Menu minor fixes

rayapetkova committed:
Update categories layouts and added new layout with image

- Added three new things in manage_portfolio
- Added a new TEST .jpg file in directory pyqt5-gui
- Added a layout for the test image

rayapetkova committed:
Update import files in login window

Imported only the things we need, and not everything from the libraries

Commit:
Categories Pixmap Fixed

    - image URL taken from database
    - image resized for max 300x100
    - image QLabel object location moved before the button

karastoyanov committed:
Categories Images Stored Locally

karastoyanov committed:
Cleaning unneccesary code

techfrapi committed:
added Story, Contact and Contributors layout

rayapetkova committed:
Update backgrounds

Made an image background for all the groupboxes and set a colour to the labels

karastoyanov committed:
some test commit

karastoyanov committed:
Merge branch 'main_menu' of github.com:Stake-And-Rope/market-app into main_menu

karastoyanov committed:
Categories Colors and Shadows Update

karastoyanov committed:
Small Visual Improvements

rayapetkova committed:
Updated categories' bckgrounds

Added a new test photo with reduced opacity and added it as a background in the category "Accessories"

rayapetkova committed:
Added new images

Found new images, reduced their opacity and added them in img --> categories_new_images

karastoyanov committed:
Code Cleaning

karastoyanov committed:
Login/Register Exec Updated

karastoyanov committed:
Subcatgories branch added

karastoyanov committed:
Food Subcategory Added

rayapetkova committed:
Deleted old images

- Deleted old images
- Renames the directory of the new images to "categories"
- Changed row 141 --> "categories_new_images" to "categories"
- Row 142: made current groupboxes rounded

karastoyanov committed:
Food Subcategory Menu

rayapetkova committed:
Update categories groupboxes --> category_name label

- Commented the lines that make the shadow
- Added a background for the category_name labels - white color
- Rounded the background of the category_name labels

karastoyanov committed:
Left Buttons Layout minor update

karastoyanov committed:
Minor GUI Updates

karastoyanov committed:
Food Subcategory Menu Update

karastoyanov committed:
Buttons Styles Update:
 - .setMaximumWidth() value updated to 150
 - custom Font set

karastoyanov committed:
Minor Design Changes
 - Description text removed
 - Objects re-alligned
 - Additional Styling to be implemented in next release

karastoyanov committed:
Food Subcategory moved to experimental folder

karastoyanov committed:
Food Subcategory moved to experimental folder

karastoyanov committed:
manage_portfolio script update

karastoyanov committed:
user info update

karastoyanov committed:
major main_menu GUI rework
- in categories groupbox - current_vertical_layout alignment set to `Bottom`
- favourites button added Top Layout
- left buttons reduced

rayapetkova committed:
Update names in subcategories

- Query needs to be fixed

karastoyanov committed:
about window minor gui rework

karastoyanov committed:
Subcategories Menu Update

karastoyanov committed:
about_window major GUI rework
 - added custom font
 - text layouts contents and design rework
 - comments updated
 - custom window icon
 - new vertical layour for github button, current version and release date
 - custom button leading to github repository

karastoyanov committed:
new custom font and github icon

karastoyanov committed:
Categories-Subcategories relationship fixed

karastoyanov committed:
subcategories/main_menu relationship fixed

karastoyanov committed:
Minor subcategories changes

rayapetkova committed:
Merge pull request #30 from Stake-And-Rope/main_menu

Merge Main Menu to Development

karastoyanov commented on pyqt5-gui/main_menu.py:
```diff
@@ -81,11 +81,13 @@ def __init__(self):
+        top_buttons_layout.addStretch(0)
+        top_buttons_layout.addSpacing(1000)
+
         top_buttons_layout.addWidget(favourites_button)
```
sdsad

karastoyanov committed:
Merge pull request #33 from Stake-And-Rope/about_window

Merge About Window in Development

karastoyanov committed:
Merge pull request #39 from Stake-And-Rope/subcats_menu

Merge Subcategories Menus in Development

rayapetkova committed:
Update subcategories menu

- Added all photos
- Made one common function in pyqt5-gui - main_menu_dev

karastoyanov committed:
Merge pull request #41 from Stake-And-Rope/raya_development

Update subcategories menu

karastoyanov committed:
major main_menu fixes

 - changes from main_menu_dev transfered to main_menu file (@rayapetkova ac3d5f8)
 - new subcategories window call with one single function
 - top layout buttons size changed (fixed heigth)
 - relantionship between files fixed (main > login/register > main_menu > subcategories)
 - code cleaning/refactoring

Co-Authored-By: Raya Petkova <122923696+rayapetkova@users.noreply.github.com>

karastoyanov committed:
Main Menu Update

karastoyanov committed:
Print Function Updated

karastoyanov committed:
Merge pull request #44 from Stake-And-Rope/raya_development

Merge `raya_development` with `experimental`

karastoyanov committed:
Merge pull request #45 from Stake-And-Rope/alex_development

Merge `alex_development` into `experimental`

karastoyanov committed:
Update Meeting Notes_week01

karastoyanov committed:
Main Menu major update
 - update user settings groupbox/layout added
 - on call, categories groupbox is being removed and user settings
   replaces it

karastoyanov committed:
user settings update

karastoyanov committed:
show/hide different groupbox on gridlayout(1,1)

karastoyanov committed:
update user settings menu fix

karastoyanov committed:
Update user data - not fully implemented

karastoyanov committed:
Fix update_user function preventing re-opening the categoroies

rayapetkova committed:
Added images and made a few changes in the subcategories_menu

- Added 3 subcategories for the category Food - Meat, Fruits, Vegetables (forgot to do it the last time)
- Added a new image for a background to the subcategories menu

karastoyanov committed:
Merge pull request #46 from Stake-And-Rope/experimental

Development Progress - week 01

karastoyanov committed:
Test file

karastoyanov committed:
update user data fully implemented

karastoyanov committed:
code refactor

karastoyanov committed:
re-open categories fixed

karastoyanov committed:
prevent re-opening any user menu

rayapetkova committed:
Updated test images, products function

karastoyanov committed:
najor fixes in products:
 - function open_products() calls products window
 - open_products() inits additional groupbox and replaces it with
   subcategories groupbox in the main_layout

rayapetkova committed:
Added some changes - deleted the groupboxes for every pduct

karastoyanov committed:
products structure updated

rayapetkova committed:
Update: new images, work with database and some changes in subcategories' file

- New images
- Added an item_description in db_handle --> manage_portfolio and changed the query because of the added item description
- Added a button "about" in subcategories.py
- Changed the query on line 231 (added the product_description and product_id to it)
- Made two deques for "products_descriptions" and "products_ids"
- One change on line 247 --> "Sunglasses" to "{cat}"

karastoyanov committed:
qplainttextedit implemented for the products description

karastoyanov committed:
Merge pull request #49 from Stake-And-Rope/experimental

Development Progress - week 02

karastoyanov committed:
PR change

karastoyanov committed:
marketapp backup - 12.06.2023

karastoyanov committed:
payment options sub-menu created

karastoyanov commented on pyqt5-gui/products.py:
```diff
@@ -0,0 +1,62 @@
+
+        product_name = products_names.popleft()
+
+        product_image = QLabel()
```
jghfdjgdf

rayapetkova committed:
Added new photos and new logic in pyqt5-gui/subcategories.py

- New photos added
- When the button for a subcategory is clicked it goes to a function (line 164) and then it calls another function which shows the subcategories menu

karastoyanov committed:
Merge pull request #53 from Stake-And-Rope/experimental

Development Progress - week 03

karastoyanov committed:
Dev week 03

karastoyanov committed:
About Window Implemented

karastoyanov committed:
edit account layout update/moved to separate fail

karastoyanov committed:
products layout update/separated in a new file

karastoyanov committed:
subcategories updated/separated in a new file

karastoyanov committed:
products window optimized. Products are generated dynamically

karastoyanov committed:
ROADMAP.md added

karastoyanov committed:
ROADMAP update

rayapetkova committed:
Added new images and removed the functions_dict in pyqt5-gui\main_menu

- New images added
- Made a new function - open_category_func(subcat_name) --> it receives the subcategory_name and return the function open_func(subcat_name)

karastoyanov committed:
Major changes in main_menu and left user buttons
 - payment options separated in a new files for better visibility and
   code optimization
 - calling the buttons from the left layout is optimized with new
   functions
 - updated open_categories function which calls back the categories
   groupbox. Needs to be double-checked and code optimized if possible

karastoyanov committed:
left buttons show/hide minor fix

rayapetkova committed:
New images added and changed labels for products

- New images added
- Changed the current_description in the file products.py to a QLabel and added current_description.setWordWrap(True) - this method makes the label multi line

rayapetkova committed:
Added new images

rayapetkova committed:
Added new images

karastoyanov committed:
minor updates

karastoyanov committed:
ROADMAP update

karastoyanov committed:
Merge pull request #54 from Stake-And-Rope/experimental

Development Progress - week 04

karastoyanov committed:
test file

rayapetkova committed:
Update - query

karastoyanov committed:
adding to favorites button implemented

Co-Authored-By: Raya Petkova rayapetkova2007@abv.com

rayapetkova committed:
Updated favourite groupbox

- main_menu.py: when the button "favourites_button" is clicked it calls the open_favourites_func. This function returns open_favourites() func. It shows the new groupbox and hides the categories_groupbox.
- favourites.py: made an inner join query which gives us all the products that "pesho" has marked as "favourite". Made all the layouts and labels for each product.

rayapetkova committed:
update

karastoyanov committed:
Scroll favorites area implemented

karastoyanov committed:
check if favorite product already exists

karastoyanov committed:
left layout show/hide major fix

karastoyanov committed:
Merge pull request #56 from Stake-And-Rope/experimental

Development Progress - week 06

karastoyanov committed:
test file

rayapetkova committed:
update test_file

karastoyanov committed:
exec permissions changed

rayapetkova committed:
Delete a product from favourites

karastoyanov committed:
payment options update

rayapetkova committed:
Added new images, implemented the top buttons layout in shorter way

- Added new images
- Made a while loop for the top buttons except the 'Log out' one, because it doesn't have functionality but there are TODOs
- Made a dictionary so it calls the function for the top buttons
- When we hover a button the cursor will change - implemented it for all the buttons in the app

nikichhh committed:
New Stylsheets

karastoyanov committed:
Conflicts resolved

rayapetkova committed:
Merge pull request #59 from Stake-And-Rope/sylsheetneshtosi

rayapetkova committed:
Added spin box, made a function for error message

- Added a spin box for every product
- When the "favourites" button is clicked it executes a query which gives us the quantity of the current product. We take the value of the current_spin_box and check if it is greater than all the available quantity of the product. If it is greater it shows an error message box which says that the user has exceeded the available quantity
- Made a function for the times when we want to show an error message box. We don't need to repeat the same code.

karastoyanov committed:
register stylesheets updated

rayapetkova committed:
Added new images and unnecessary lines in main_menu.py file

rayapetkova committed:
Added new images

rayapetkova committed:
Added images (there are no more)

nikichhh committed:
Style changes for login and register menus

nikichhh committed:
Enhance styles.qss

All of the stylesheets from login.py and register.py have been moved to style.qss.

karastoyanov committed:
Merge pull request #58 from Stake-And-Rope/experimental

Development Progress - week 07

karastoyanov committed:
test file for PR

karastoyanov committed:
Major Fixes on Login and Register Menus
 - files relantions fixed and reworked
 - new login backed logic
 - database admin and client init functions from postgres_conn
   functionality is optimized. Clients are executed only once when
   starting the app

karastoyanov committed:
Update ROADMAP.md

karastoyanov committed:
code rework

karastoyanov committed:
repo dir strucutre optimized

karastoyanov committed:
Postgres Dump Backup - 04.08

karastoyanov committed:
Dump Backup

karastoyanov committed:
Dump backup

rayapetkova committed:
When the 'favourites' button is clicked it shows the user's favourites items with default value (spin box)

- Added a new column 'quantity_wanted' in favourite_products' table which shows what quantity the user wants
- Implemented a QSpinBox in 'favourites.py' file which has a default value. The value comes from the inner join query (added a 'favourite_products.quantity_wanted' so we can get the wanted quantity)

rayapetkova committed:
Basic implementation of the 'basket' logic

karastoyanov committed:
typo fix

karastoyanov committed:
register form fixed

karastoyanov committed:
pop-out window on approved registration

karastoyanov committed:
register_user new logic

karastoyanov committed:
dynamic login fixed

karastoyanov committed:
fix dynamic login query for currently logged in user in main_menu

karastoyanov committed:
updated relations && fixed update user info functionality

karastoyanov committed:
products update

karastoyanov committed:
code cleaning and optimization

karastoyanov committed:
conflicts resolved partially

karastoyanov committed:
git revert

karastoyanov committed:
favorites and main_menu conflicts fixed

karastoyanov committed:
products conflicts fixed

karastoyanov committed:
Merge pull request #63 from Stake-And-Rope/files-assembly

Files assembly

karastoyanov committed:
final touches after conflicts resolving

karastoyanov committed:
basket sql query fix

rayapetkova committed:
Basket logic fixed

nikichhh committed:
Show password button logic and displayed added

The "Show password" button is for now only in the login.py and register.py files

nikichhh committed:
Log Out button added in the main_menu file

"Log Out" button logic and displayed added in the main_menu.py file

rayapetkova committed:
Update basket/favourites

karastoyanov committed:
minor changes

karastoyanov committed:
minor changes

karastoyanov committed:
payment options menu reworked

karastoyanov committed:
payment options update

karastoyanov committed:
DeleteClass fix

karastoyanov committed:
Create New Card labels created

karastoyanov committed:
payment options minor update

karastoyanov committed:
payment options delete card functionality fixed

karastoyanov committed:
add new payment option implemented

karastoyanov committed:
code review messages

nikichhh committed:
Added after clicking "Enter"/"Return" to do the same as clicked with mouse

I overrided the keyPressEvent function making the log_in_button to respond after clicking "Enter" or "Return" key (depends on the OS)

nikichhh committed:
Visual changes on some files

Small visual changes in the files: main_menu, subcategories, products, basket and favourites

rayapetkova committed:
Remove current spin box option in products menu (main)

karastoyanov committed:
hiding favorites and basket menu's fixed

karastoyanov committed:
remove favorites button - functinality not implemented

rayapetkova committed:
Update some functionalities

- The user can add a product from favourites to basket and from basket to favourites
- Added a remove button to the products in the basket
- Added a ToolTip to every button that refers to a product

karastoyanov committed:
Project Update:
 - 'Confirm Order' button added in basket menu
 - code optimizing and cleaning
 - test function for confirming the order -> confirm_order.py in
   db_handle dir

rayapetkova committed:
Confirm order (implemented button functionality)

karastoyanov committed:
conflicts fix

karastoyanov committed:
Merge pull request #60 from Stake-And-Rope/experimental

Development progress - week 10

karastoyanov commented:
Closing this PR due to missing the deadline. We will keep this beta in the main branch and...hopefully next year we will be able to finish some of the last planned features. 

I will prepare short changelog and release in next few days.
